### PR TITLE
WIP: Update book tutorials to Pumas 2.0

### PR DESCRIPTION
**Pull request by vjd**
Update all tutorials to Pumas 2.0
Most updates are for plotting frameworks.

### CODE REVIEW

File: tutorials/PKPDDataAnalysisBook/PK/pk19.jmd [hunk 1]
```diff
@@ -115,51 +134,52 @@ param = ( tvvc        = 1.06405,
...
+        linewidth = 4,
+        axis = (xlabel = "Time (minute)", 
+                ylabel = "PK19 Concentrations (μmol/L)",
+                xticks = 0:50:300, yscale = log10))
+sim_plot!(ax, pk_19, 
+              sim_pop3_sub,
+              observations=:met, 
+              color = :redsblues, linewidth = 4)                
+axislegend(ax) 
+fig
```
Comment (vjd) [inline tutorials/PKPDDataAnalysisBook/PK/pk19.jmd:180]: @MichaelHatherly can you check why this overlay on the same axis errors? Currently, with the code above I get 

```julia
julia> sim_plot!(ax, pk_19, 
                     sim_pop3_sub,
                     observations=:met, 
                     color = :redsblues, linewidth = 4)
ERROR: MethodError: no method matching subject_ids(::PumasModel{ParamSet{NamedTuple{(:tvvc, :tvvp, :tvq, :tvvmax, :tvkm, :tvkme, :tvvme, :Ω, :σ_prop_cp, :σ_prop_met), Tuple{RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, PDiagDomain{PDMats.PDiagMat{Float64, Vector{Float64}}}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}}}}, var"#273#297", var"#274#298", var"#276#300", var"#278#302", ODEProblem{Nothing, Tuple{Nothing, Nothing}, false, Nothing, ODEFunction{false, ModelingToolkit.ODEFunctionClosure{var"#279#303", var"#280#304"}, LinearAlgebra.UniformScaling{Bool}, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Vector{Symbol}, Symbol, typeof(SciMLBase.DEFAULT_OBSERVED), Nothing}, Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}, SciMLBase.StandardODEProblem}, var"#281#305", var"#289#313", ModelingToolkit.ODESystem})
Closest candidates are:
  subject_ids(::Vector{T} where T) at /Users/vijay/.julia/packages/PumasPlots/IcfzO/src/recipes/subject_fits.jl:167
  subject_ids(::Pumas.AbstractFittedPumasModel) at /Users/vijay/.julia/packages/PumasPlots/IcfzO/src/recipes/subject_fits.jl:168
  subject_ids(::Pumas.FittedPumasModelInspection) at /Users/vijay/.julia/packages/PumasPlots/IcfzO/src/recipes/subject_fits.jl:169
  ...
Stacktrace:
 [1] plot!(p::Combined{PumasPlots.sim_plot, Tuple{PumasModel{ParamSet{NamedTuple{(:tvvc, :tvvp, :tvq, :tvvmax, :tvkm, :tvkme, :tvvme, :Ω, :σ_prop_cp, :σ_prop_met), Tuple{RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, PDiagDomain{PDMats.PDiagMat{Float64, Vector{Float64}}}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}}}}, var"#273#297", var"#274#298", var"#276#300", var"#278#302", ODEProblem{Nothing, Tuple{Nothing, Nothing}, false, Nothing, ODEFunction{false, ModelingToolkit.ODEFunctionClosure{var"#279#303", var"#280#304"}, LinearAlgebra.UniformScaling{Bool}, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Vector{Symbol}, Symbol, typeof(SciMLBase.DEFAULT_OBSERVED), Nothing}, Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}, SciMLBase.StandardODEProblem}, var"#281#305", var"#289#313", ModelingToolkit.ODESystem}, Vector{Pumas.SimulatedObservations{Subject{NamedTuple{(:cp, :met), Tuple{Nothing, Nothing}}, Pumas.ConstantCovar{NamedTuple{(), Tuple{}}}, Vector{Pumas.Event{Float64, Float64, Float64, Float64, Float64, Float64, Int64}}, Nothing}, StepRange{Int64, Int64}, NamedTuple{(:cp, :dv_cp, :met, :dv_met), NTuple{4, Vector{Float64}}}}}}})
   @ PumasPlots ~/.julia/packages/PumasPlots/IcfzO/src/recipes/sim_plot.jl:51
 [2] plot!(scene::Scene, P::Type{Combined{PumasPlots.sim_plot, Tuple{PumasModel{ParamSet{NamedTuple{(:tvvc, :tvvp, :tvq, :tvvmax, :tvkm, :tvkme, :tvvme, :Ω, :σ_prop_cp, :σ_prop_met), Tuple{RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, PDiagDomain{PDMats.PDiagMat{Float64, Vector{Float64}}}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}}}}, var"#273#297", var"#274#298", var"#276#300", var"#278#302", ODEProblem{Nothing, Tuple{Nothing, Nothing}, false, Nothing, ODEFunction{false, ModelingToolkit.ODEFunctionClosure{var"#279#303", var"#280#304"}, LinearAlgebra.UniformScaling{Bool}, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Vector{Symbol}, Symbol, typeof(SciMLBase.DEFAULT_OBSERVED), Nothing}, Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}, SciMLBase.StandardODEProblem}, var"#281#305", var"#289#313", ModelingToolkit.ODESystem}, Vector{Pumas.SimulatedObservations{Subject{NamedTuple{(:cp, :met), Tuple{Nothing, Nothing}}, Pumas.ConstantCovar{NamedTuple{(), Tuple{}}}, Vector{Pumas.Event{Float64, Float64, Float64, Float64, Float64, Float64, Int64}}, Nothing}, StepRange{Int64, Int64}, NamedTuple{(:cp, :dv_cp, :met, :dv_met), NTuple{4, Vector{Float64}}}}}}}}, attributes::Attributes, input::Tuple{Observable{PumasModel{ParamSet{NamedTuple{(:tvvc, :tvvp, :tvq, :tvvmax, :tvkm, :tvkme, :tvvme, :Ω, :σ_prop_cp, :σ_prop_met), Tuple{RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, PDiagDomain{PDMats.PDiagMat{Float64, Vector{Float64}}}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}}}}, var"#273#297", var"#274#298", var"#276#300", var"#278#302", ODEProblem{Nothing, Tuple{Nothing, Nothing}, false, Nothing, ODEFunction{false, ModelingToolkit.ODEFunctionClosure{var"#279#303", var"#280#304"}, LinearAlgebra.UniformScaling{Bool}, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Vector{Symbol}, Symbol, typeof(SciMLBase.DEFAULT_OBSERVED), Nothing}, Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}, SciMLBase.StandardODEProblem}, var"#281#305", var"#289#313", ModelingToolkit.ODESystem}}, Observable{Vector{Pumas.SimulatedObservations{Subject{NamedTuple{(:cp, :met), Tuple{Nothing, Nothing}}, Pumas.ConstantCovar{NamedTuple{(), Tuple{}}}, Vector{Pumas.Event{Float64, Float64, Float64, Float64, Float64, Float64, Int64}}, Nothing}, StepRange{Int64, Int64}, NamedTuple{(:cp, :dv_cp, :met, :dv_met), NTuple{4, Vector{Float64}}}}}}}, args::Observable{Tuple{PumasModel{ParamSet{NamedTuple{(:tvvc, :tvvp, :tvq, :tvvmax, :tvkm, :tvkme, :tvvme, :Ω, :σ_prop_cp, :σ_prop_met), Tuple{RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, PDiagDomain{PDMats.PDiagMat{Float64, Vector{Float64}}}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}}}}, var"#273#297", var"#274#298", var"#276#300", var"#278#302", ODEProblem{Nothing, Tuple{Nothing, Nothing}, false, Nothing, ODEFunction{false, ModelingToolkit.ODEFunctionClosure{var"#279#303", var"#280#304"}, LinearAlgebra.UniformScaling{Bool}, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Vector{Symbol}, Symbol, typeof(SciMLBase.DEFAULT_OBSERVED), Nothing}, Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}, SciMLBase.StandardODEProblem}, var"#281#305", var"#289#313", ModelingToolkit.ODESystem}, Vector{Pumas.SimulatedObservations{Subject{NamedTuple{(:cp, :met), Tuple{Nothing, Nothing}}, Pumas.ConstantCovar{NamedTuple{(), Tuple{}}}, Vector{Pumas.Event{Float64, Float64, Float64, Float64, Float64, Float64, Int64}}, Nothing}, StepRange{Int64, Int64}, NamedTuple{(:cp, :dv_cp, :met, :dv_met), NTuple{4, Vector{Float64}}}}}}})
   @ Makie ~/.julia/packages/Makie/xbI6d/src/interfaces.jl:428
 [3] plot!(::Scene, ::Type{Combined{PumasPlots.sim_plot, ArgType} where ArgType}, ::Attributes, ::PumasModel{ParamSet{NamedTuple{(:tvvc, :tvvp, :tvq, :tvvmax, :tvkm, :tvkme, :tvvme, :Ω, :σ_prop_cp, :σ_prop_met), Tuple{RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, PDiagDomain{PDMats.PDiagMat{Float64, Vector{Float64}}}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}}}}, var"#273#297", var"#274#298", var"#276#300", var"#278#302", ODEProblem{Nothing, Tuple{Nothing, Nothing}, false, Nothing, ODEFunction{false, ModelingToolkit.ODEFunctionClosure{var"#279#303", var"#280#304"}, LinearAlgebra.UniformScaling{Bool}, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Vector{Symbol}, Symbol, typeof(SciMLBase.DEFAULT_OBSERVED), Nothing}, Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}, SciMLBase.StandardODEProblem}, var"#281#305", var"#289#313", ModelingToolkit.ODESystem}, ::Vararg{Any, N} where N; kw_attributes::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ Makie ~/.julia/packages/Makie/xbI6d/src/interfaces.jl:339
 [4] plot!(::Scene, ::Type{Combined{PumasPlots.sim_plot, ArgType} where ArgType}, ::Attributes, ::PumasModel{ParamSet{NamedTuple{(:tvvc, :tvvp, :tvq, :tvvmax, :tvkm, :tvkme, :tvvme, :Ω, :σ_prop_cp, :σ_prop_met), Tuple{RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, PDiagDomain{PDMats.PDiagMat{Float64, Vector{Float64}}}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}}}}, var"#273#297", var"#274#298", var"#276#300", var"#278#302", ODEProblem{Nothing, Tuple{Nothing, Nothing}, false, Nothing, ODEFunction{false, ModelingToolkit.ODEFunctionClosure{var"#279#303", var"#280#304"}, LinearAlgebra.UniformScaling{Bool}, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Vector{Symbol}, Symbol, typeof(SciMLBase.DEFAULT_OBSERVED), Nothing}, Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}, SciMLBase.StandardODEProblem}, var"#281#305", var"#289#313", ModelingToolkit.ODESystem}, ::Vector{Pumas.SimulatedObservations{Subject{NamedTuple{(:cp, :met), Tuple{Nothing, Nothing}}, Pumas.ConstantCovar{NamedTuple{(), Tuple{}}}, Vector{Pumas.Event{Float64, Float64, Float64, Float64, Float64, Float64, Int64}}, Nothing}, StepRange{Int64, Int64}, NamedTuple{(:cp, :dv_cp, :met, :dv_met), NTuple{4, Vector{Float64}}}}})
   @ Makie ~/.julia/packages/Makie/xbI6d/src/interfaces.jl:308
 [5] plot!(::Axis, ::Type{Combined{PumasPlots.sim_plot, ArgType} where ArgType}, ::Attributes, ::PumasModel{ParamSet{NamedTuple{(:tvvc, :tvvp, :tvq, :tvvmax, :tvkm, :tvkme, :tvvme, :Ω, :σ_prop_cp, :σ_prop_met), Tuple{RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, PDiagDomain{PDMats.PDiagMat{Float64, Vector{Float64}}}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}, RealDomain{Int64, TransformVariables.Infinity{true}, Int64}}}}, var"#273#297", var"#274#298", var"#276#300", var"#278#302", ODEProblem{Nothing, Tuple{Nothing, Nothing}, false, Nothing, ODEFunction{false, ModelingToolkit.ODEFunctionClosure{var"#279#303", var"#280#304"}, LinearAlgebra.UniformScaling{Bool}, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Vector{Symbol}, Symbol, typeof(SciMLBase.DEFAULT_OBSERVED), Nothing}, Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}, SciMLBase.StandardODEProblem}, var"#281#305", var"#289#313", ModelingToolkit.ODESystem}, ::Vararg{Any, N} where N; kw_attributes::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ Makie.MakieLayout ~/.julia/packages/Makie/xbI6d/src/makielayout/layoutables/axis.jl:658
 [6] plot!
   @ ~/.julia/packages/Makie/xbI6d/src/makielayout/layoutables/axis.jl:653 [inlined]
 [7] #plot!#318
   @ ~/.julia/packages/Makie/xbI6d/src/makielayout/layoutables/axis.jl:670 [inlined]
 [8] sim_plot!(::Axis, ::Vararg{Any, N} where N; attributes::Base.Iterators.Pairs{Symbol, Any, Tuple{Symbol, Symbol, Symbol}, NamedTuple{(:observations, :color, :linewidth), Tuple{Symbol, Symbol, Int64}}})
   @ PumasPlots ~/.julia/packages/MakieCore/S8PkO/src/recipes.jl:35
 [9] top-level scope
   @ REPL[201]:1
```

File: tutorials/PKPDDataAnalysisBook/PK/pk16.jmd [hunk 2]
```diff
@@ -138,7 +155,8 @@ To simulate the following for the above subject with specific observation time p
 
 ```julia; results="hidden"
 Random.seed!(123)
-sim_sub1 = simobs(pk_16,sub1,param,obstimes = [0.5,1,2,4,6.1,7.6,8.02,12.05,12.15,15.95,22.13,23.89,24.05,24.46,24.94,25.94,26.96,27.95,29.97,31.94,35.96,36.2,48,48.2,54,60,60.2,72,72.2])
+sim_sub1 = simobs(pk_16, sub1, param,
```
Comment (vjd) [inline tutorials/PKPDDataAnalysisBook/PK/pk16.jmd:158]: @andreasnoack  or @pkofod can one of you check what is wrong with this. `solve` on this model gives NAN's in Pumas 2.0 but not in Pumas 1.1.

File: tutorials/PKPDDataAnalysisBook/PK/pk19.jmd (hunk 1 above)
Comment (MichaelHatherly) [reply tutorials/PKPDDataAnalysisBook/PK/pk19.jmd:180]: The docstring for `sim_plot!` need updating to say it only takes the population, not the model as well. Or we need to adjust the code to handle taking both a model and population in the `!` versions of these functions, even though model isn't used anywhere in the method.

Comment (MichaelHatherly) [reply tutorials/PKPDDataAnalysisBook/PK/pk19.jmd:180]: https://github.com/PumasAI/PumasPlots.jl/issues/137

File: tutorials/PKPDDataAnalysisBook/PK/pk27.jmd [hunk 3]
```diff
@@ -73,7 +88,10 @@ pk_27           = @model begin
     Kout        = tvkout * exp(η[7])
     Kerl        = tvkerl * exp(η[8])
     Vc          = tvvc * exp(η[9])
-    bioav       = (Central = 1/Vc,)
+  end
+
+  @dosecontrol begin
+    bioav       = (Central = 1/Vc_,)
```
Comment (vjd) [inline tutorials/PKPDDataAnalysisBook/PK/pk27.jmd:94]: @pkofod can you please help here. How do I bring in `Vc` from the `pre` block into the `@dosecontrol` block?

Comment (pkofod) [reply tutorials/PKPDDataAnalysisBook/PK/pk27.jmd:94]: You'd just have to re-specify it. Just copy that line down.

**Comment by vjd** [comment]
pk5.jmd - works in Pumas 1.1, but not in 2.0
pk6.jmd - works in Pumas 1.1, but not in 2.0
pk15.jmd - remove
pk16.jmd - works in Pumas 1.1, but not in 2.0
pk40.jmd - works in Pumas 1.1, but not in 2.0

File: tutorials/PKPDDataAnalysisBook/PK/pk27.jmd (hunk 3 above)
Comment (vjd) [reply tutorials/PKPDDataAnalysisBook/PK/pk27.jmd:95]: done

**Pull request merged**